### ISSUE TITLE
"photoswipe-gallery" style interferes with views style plugin

### ISSUE THREAD

**Issue by korontari** [backdrop-contrib/photoswipe#17]
Fist of all thank you very much for your work and support of the Photoswipe module!

After recent update I found that the module breaks styles for some views style plugins ([Mosaic grid](https://github.com/backdrop-contrib/mosaic_grid) module to be specific). I have noticed that the added ".photoswipe-gallery" class adds margin and alignment and looks like this:
```
.photoswipe-gallery div {
  float: left;
  margin-right: 5px;
}
```
If I make this CSS class empty (_.photoswipe-gallery div {}_) then Photoswipe still recognizes the view as a gallery but it doesn't breaks any styles used by views style plugins. Let me know what you think and if you see any other/better solutions for this.

**Comment by Graham-72** [comment]
@korontari thanks for reporting this issue. I have been investigating a related and perhaps even more significant issue. The template that I added in PR #8 is actually being applied to any view whether or not the view is designed to use Photoswipe, and adding the class `photoswipe-gallery` where it is not needed and thus causes problems. 

I have found difficulty in identifying a fix for this because it needs a way of identifying that a view is of type 'photoswipe' but I have found there is a such a value 'type' in the image field settings for the view. I will shortly provide a PR for modifying the  `views-view.tpl.php` I have added to photoswipe.

I will also try it in combination with the Mosaic grid module, something I have not tried before. If there are still problems I may have to abandon my idea of getting the module to automatically add the necessary class `photoswipe-gallery` when used in a view.

**Comment by Graham-72** [comment]
@korontari I think my latest commit fixes this. I found that when using Mosaic Grid and setting the image formatter to be photoswipe the template was incorrectly setting the view as a photoswipe gallery. Please kindly check that this clash is now avoided. If you confirm this I will publish a new release. Thank you.

Added : I have found another small problem and will revise the PR. Also, a question, do you expect to be able to use the mosaic grid together with a photoswipe gallery? At the moment something in the Javascript is over-riding the styles.

**Comment by korontari** [comment]
@Graham-72 , thank you for the fix! I should be able to check it tomorrow and get back to you with the results.

Regarding your question. Yes, I am using Photoswipe with the Mosaic Grid. I am the maintainer of the Mosaic Grid :) and if you find there is an issue with it please let me know, I can check it. But so far it worked fine with other image formatters (Featherlight for example).

**Comment by Graham-72** [comment]
@korontari I agree with your issue that the added 'photoswipe-gallery' class is causing the problem and should be removed.

**Comment by Graham-72** [comment]
Hopefully this PR #23  finally fixes the issue. It also fixes another problem I was seeing on one of my sites.

**Comment by korontari** [comment]
I have investigated a bit into the issue regarding adding the photoswipe-gallery style to the views.
It should be taken into consideration that a view can have several displays, not only the default one. Also I noticed that some data is not always present in config files and taking info about fields from there does not always work.

I found that taking info about fields from views in template preprocessing function(template_preprocess_views_view) provides more consistent result. So I tried to do that in PR #24. It does roughly the same as you did in views template, but before the view is processed. When I tested, this worked correctly on a view with several displays(pages) and different image formatters on each page. Would you kindly check it please?

UPD: I have just checked the PR #23 and noticed that photoswipe-gallery class is removed from the default-skin.css. That actually  solves the issue with Mosaic Grid, it now works fine! I believe if you test #24 and it works, the issue can be closed. 😃 

UPD2: Just an afterthought and idea for a future update. Ideally would be great to offer user to choose if he wants the image fields in view to be regarded as a gallery or separate entities. Maybe add checkmark 'Display images in a gallery' in image formatter settings?

**Comment by Graham-72** [comment]
Thank you for PR #24 
I like the idea (UPD2) of including a setting for opting not to have images displayed as a gallery. I have written some code to add this and will produce a PR as a suggestion. I would like to include this before issuing a new release.

**Comment by korontari** [comment]
@Graham-72 , thank you! I would be glad to test a PR with "gallery option".

**Comment by Graham-72** [comment]
> @Graham-72 , thank you! I would be glad to test a PR with "gallery option".

@korontari please see PR #25 . I have tried to do this in a way that does not interfere with any existing uses of photoswipe. The inhibit setting will only be added and effective when a view is updated or a display setting is changed. Hence I have chosen to make it an inhibit. Also, for the case where a view has several displays, I have assumed a separate setting is needed for each.

**Comment by korontari** [comment]
@Graham-72 , thank you! I have tested the PR #25 in my environment and it seems to work properly. Tested with multiple displays on one view as well, everything seems working fine! The "Inhibit gallery action" does sound to me a bit counter-intuitive, a kind of rocket science language. It seems to me more straightforward and user friendly just to name it "Display as gallery" and set it on by default. But as English is not my native language I am not firm in my opinion, just giving it as a personal feedback. 😄
